src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -59,81 +52,63 @@ public function __construct(ObjectManager $manager, string $class, IdReader $idR
     /**
      * {@inheritdoc}
      */
-    public function loadChoiceList(callable $value = null)
+    protected function loadChoices(): iterable
     {
-        if ($this->choiceList) {
-            return $this->choiceList;
-        }
-
-        $objects = $this->objectLoader
+        return $this->objectLoader
             ? $this->objectLoader->getEntities()
-            : $this->manager->getRepository($this->class)->findAll();
-
-        return $this->choiceList = new ArrayChoiceList($objects, $value);
+            : $this->manager->getRepository($this->class)->findAll()
+        ;
     }
 
     /**
-     * {@inheritdoc}
+     * This method is here for legacy.
+     *
+     * To remove in 6.0
```

Comment:
`@internal to be removed in Symfony 6` in place of the whole content of the docblock to me